docs/arch-split.md

```diff
@@ -687,6 +746,39 @@ generates limited duplication: a fact from `Foo_AI_1` will be duplicated in
 `Foo_AI_2`, but not in `Foo_AI_3+`.
 
 
+### Temporarily proving a fact in the Arch locale
+
+The concept of "generic consequences of architecture-specific properties" shows
+up in a few places. Normally, as outlined above, we prefer either exporting
+enough facts to prove the property in the generic context or requiring the
+property as a locale assumption. However, sometimes we end up in a situation
+where the same proof will work on all architectures and spelling it out with
+locale assumptions is inconvenient. For example (from `Invariants_AI`):
+
+```isabelle
+(* generic consequence of architecture-specific details *)
+lemma (in Arch) valid_arch_cap_pspaceI:
+  "⟦ valid_arch_cap acap s; kheap s = kheap s' ⟧ ⟹ valid_arch_cap acap s'"
+  unfolding valid_arch_cap_def
+  by (auto intro: obj_at_pspaceI split: arch_cap.split)
+
+requalify_facts Arch.valid_arch_cap_pspaceI
+```
+
+In this case, no matter what the architecture, the `valid_arch_cap` function
+will only ever look at the heap, so this proof will always work.
+
+There are some considerations when using this strategy:
+
+1. We use the Arch locale without a `global_naming`, as its performance better
+   than entering the Arch locale and proving the lemma there. This means its
```

Comment:
```suggestion
1. We use the Arch locale without `global_naming`, as its performance is better
   than entering the Arch locale and proving the lemma there. This means its
```
```suggestion
1. We use the Arch locale without a `global_naming`, as its performance better
   than entering the Arch locale and proving the lemma there. This means its
```

Reply:
Did something go wrong on the second one? It says "The suggestion is invalid because no changes were made to the code"

Reply:
Nothing wrong on that line, I was just trying to provide sentence context. Did not know that this would lead to failure, sorry.

Reply:
Thanks, fixed